tests/Utils/UnifiedDiffAssertTraitIntegrationTest.php

```diff
@@ -0,0 +1,129 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of sebastian/diff.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SebastianBergmann\Diff\Utils;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Process\Process;
+
+/**
+ * @requires OS Linux
+ *
+ * @coversNothing
+ */
+final class UnifiedDiffAssertTraitIntegrationTest extends TestCase
+{
+    use UnifiedDiffAssertTrait;
+
+    private $filePatch;
+
+    protected function setUp()
+    {
+        $this->filePatch = __DIR__ . '/../fixtures/out/patch.txt';
+
+        $this->cleanUpTempFiles();
+    }
+
+    /**
+     * @param string $fileFrom
+     * @param string $fileTo
+     *
+     * @dataProvider provideFilePairsCases
+     */
+    public function testValidPatches(string $fileFrom, string $fileTo)
+    {
+        $command = \sprintf(
+            'diff -u %s %s > %s',
+            \escapeshellarg(\realpath($fileFrom)),
+            \escapeshellarg(\realpath($fileTo)),
+            \escapeshellarg($this->filePatch)
+        );
+
+        $p = new Process($command);
+        $p->run();
+
+        $exitCode = $p->getExitCode();
+
+        if (0 === $exitCode) {
+            // odd case when two files have the same content. Test after executing as it is more efficient than to read the files and check the contents every time.
+            $this->addToAssertionCount(1);
+
+            return;
+        }
+
+        $this->assertSame(
+            1, // means `diff` found a diff between the files we gave it
+            $exitCode,
+            \sprintf(
+                "Command exec. was not successful:\n\"%s\"\nOutput:\n\"%s\"\nStdErr:\n\"%s\"\nExit code %d.\n",
+                $command,
+                $p->getOutput(),
+                $p->getErrorOutput(),
+                $p->getExitCode()
+            )
+        );
+
+        $this->assertValidUnifiedDiffFormat(\file_get_contents($this->filePatch));
```

Comment:
why sometimes there is native `file_get_contents` used, and sometimes `getFileContent` wrapper ?

Reply:
unified in all tests now